doc/sphinxman/source/build_planning.rst

```diff
@@ -0,0 +1,1152 @@
+.. run ``gcc -v`` to find out what version of GCC you're using, and then
+.. download the corresponding GFortran from
+.. <http://r.research.att.com/tools/>.  If you configure Psi on a Mac without
+.. any Fortran compiler it will set itself up correctly, so this is only
+.. necessary if you want a Fortran compiler for other purposes.
+
+
+.. _`cmake:python`:
+
+How to configure Python for building Psi4
+-----------------------------------------
+
+**Role and Dependencies**
+
+* Role |w---w| In |PSIfour|, Python allows the core compiled C++ code to
+  be flexibly accessed for manipulation and extension in an interpreted
+  language.
+
+* Downstream Dependencies |w---w| |PSIfour| |dr| Python Interpreter
+
+**CMake Variables**
+
+* :makevar:`PYTHON_EXECUTABLE` |w---w| specify name or full path to Python interpreter.
+* :makevar:`PYTHON_LIBRARY` |w---w| specify path to Python library.
+* :makevar:`PYTHON_INCLUDE_DIR` |w---w| specify directory of Python headers. Contains ``Python.h``.
+
+**Examples**
+
+A. Build with detected Python from :envvar:`PATH`
+
+  .. code-block:: bash
+
+    >>> cmake
+
+B. Build with specific Python
+
+  .. code-block:: bash
+
+    >>> cmake -DPYTHON_EXECUTABLE=/path/to/interp/python2.7
+
+C. Build with full Python specification to root directory ``${PFXC}``
+
+  .. code-block:: bash
+
+    >>> cmake -DPYTHON_EXECUTABLE="${PFXC}/bin/python" \
+              -DPYTHON_LIBRARY="${PFXC}/lib/libpython3.5m.so" \
+              -DPYTHON_INCLUDE_DIR="${PFXC}/include/python3.5m"
+
+
+.. _`faq:runtimepython`:
+
+What Python is Psi4 running
+---------------------------
+
+The Python detected at build-time is embedded into the |PSIfour|
+executable. That is, the top line of ``bin/psi4`` is something like
+``#!/path/to/miniconda/envs/p4deps/bin/python3.5``, and that's the Python
+through which |PSIfour| is running, *not* the Python of ``which python``.
+To use a different Python with |PSIfour| in the short term, just
+``path/to/desired/python psi4`` on the command line to override the
+shebang line. To use a different Python with |PSIfour| in the long term,
+edit the shebang line.
+
+If you're using |PSIfour| as a Python module, then |PSIfour| *is* running
+the Python of ``which python``.
+
+
+.. _`faq:wrongpyfalse`:
+
+How to fix "``undefined symbol: _Py_FalseStruct``"
+--------------------------------------------------
+
+You're probably loading a Py3-compiled Psi4 in Py2. Switch interpreters
+and re-run. A python of proper Py2 or Py3-ness is baked into the |PSIfour|
+"executable", so you'll see this error only for Psi4 as Python module.
+
+
+.. _`faq:gdblldb`:
+
+How to use ``gdb`` and ``lldb`` with Psi4
+-----------------------------------------
+
+Debugging |PSIfour| has gotten a little confusing now that it's running through Python. Here's the syntax ::
+
+  >>> cd {objdir}
+  >>> lldb -- python stage/{prefix}/bin/psi4 ../tests/tu1-h2o-energy/input.dat
+  >>> (lldb) run
```

Comment:
Thought the ``--`` was how you got args in on the command line, instead of having to split the normal command into lldb-launch and run-rest-of-args steps. From [here](http://lldb.llvm.org/lldb-gdb.html). Maybe can do same with gdb.